plugins/com.gwtplugins.gdt.eclipse.suite/src/com/google/gdt/eclipse/suite/wizards/NewWebAppTemplateProjectWizardPage.java

```diff
@@ -0,0 +1,532 @@
+  private Label outDirLabel;
+  private Text outDirText;
+  private Button outDirWorkspaceButton;
+  private Text packageText;
+  private Text projectNameText;
+  private Button useGwtCheckbox;
+  private Combo templateCombo;
+  private final String workspaceDirectory;
+  private List<ProjectTemplate> templates;
+
+  public NewWebAppTemplateProjectWizardPage() {
+    super("createProject");
+    setTitle("Create a Web Application Project");
+    setDescription("Create a Web Application project in the workspace or in an external location");
+
+    IWorkspaceRoot workspaceRoot = ResourcesPlugin.getWorkspace().getRoot();
+    workspaceDirectory = workspaceRoot.getLocation().toOSString();
+
+    for (IProject project : workspaceRoot.getProjects()) {
+      existingProjectNames.add(project.getName());
+    }
+  }
+
+  @Override
+  public void createControl(Composite parent) {
+
+    final ScrolledComposite scroller = new ScrolledComposite(parent, SWT.H_SCROLL | SWT.V_SCROLL);
+
+    final GridLayout gridLayout = new GridLayout();
+
+    Composite containerOfComponents = new Composite(scroller, SWT.NULL);
+    containerOfComponents.setLayout(gridLayout);
+
+    setControl(containerOfComponents);
+
+    // TODO: convert these fields to use StringDialogField instead?
+
+    // Project name
+    final Label projectNameLabel = new Label(containerOfComponents, SWT.NONE);
+    projectNameLabel.setText("Project name:");
+
+    projectNameText = new Text(containerOfComponents, SWT.BORDER);
+    final GridData gd1 = new GridData(GridData.FILL_HORIZONTAL);
+    gd1.horizontalSpan = 2;
+    projectNameText.setLayoutData(gd1);
+    projectNameText.addModifyListener(new ModifyListener() {
+      @Override
+      public void modifyText(ModifyEvent e) {
+        updateControls();
+      }
+    });
+
+    // Package
+    final Label packageLabel = new Label(containerOfComponents, SWT.NONE);
+    packageLabel.setText("Package: (e.g. com.example.myproject)");
+
+    packageText = new Text(containerOfComponents, SWT.BORDER);
+    final GridData gd2 = new GridData(GridData.FILL_HORIZONTAL);
+    gd2.horizontalSpan = 2;
+    packageText.setLayoutData(gd2);
+    packageText.addModifyListener(new ModifyListener() {
+      @Override
+      public void modifyText(ModifyEvent e) {
+        updateControls();
+      }
+    });
+    final Label templateLabel = new Label(containerOfComponents, SWT.NONE);
+    templateLabel.setText("Select Template:");
+    templateCombo = new Combo(containerOfComponents, SWT.NONE);
+    GridData gd3 = new GridData(GridData.FILL_HORIZONTAL);
+    gd3.horizontalSpan = 2;
+    templateCombo.setLayoutData(gd3);
+    try {
+      templates = ProjectTemplate.getTemplates();
+      Collections.sort(templates);
+      for(int i=0;i<templates.size();i++)
+      {
+        ProjectTemplate templ = templates.get(i);
+        templateCombo.add(templ.getName());
+      }
+    } catch (IOException e1) {
+      e1.printStackTrace();
```

Comment:
probably should throw or return early so we halt the process? it seems bad to stop with some of the templates applied but not all, and no negative feedback (except this trace in the logs)